static/src/javascripts/projects/commercial/modules/dfp/Advert.ts

```diff
@@ -81,8 +55,21 @@ class Advert {
 	hasPrebidSize = false;
 	lineItemId: number | null = null;
 
-	constructor(adSlotNode: HTMLElement) {
-		const sizes = getAdBreakpointSizes(adSlotNode);
+	constructor(adSlotNode: HTMLElement, additionalSizes?: SizeMapping) {
+		const sizes = getAdSizeMapping(adSlotNode.dataset.name ?? '') ?? {};
```

Comment:
Would the empty string return a default set of sizes?

Reply:
what would the default be?

Reply:
We had a set of 'commonSizeMappings' but I see that those have been removed - which is great as it simplifies the size mappings logic!

I'm open to ideas but we have the options:

- this is an error case so we log and do nothing
- we pass the equivalent of empty sizes to `defineSlot`
- we default to an MPU

Happy to defer to what you think is the best option.

Also would turning the condition inside out help i.e.:

```suggestion
		const sizes = adSlotNode.dataset.name ? getAdSizeMapping(adSlotNode.dataset.name) : {};
```

Reply:
Error log is a good shout, maybe even throw so it's picked up by sentry?

I guess this is akin to no data attributes being present on the slot using the old method, which I don't think was handled.

I think play it safe and not show a potentially janky ad?

Reply:
An exception is a good idea as an unmatched slot name shouldn't happen so having visibility would very helpful.